### PR TITLE
[main] Migrate auth/providers from Norman to Wrangler

### PR DESCRIPTION
**Pull request by crobby**
## Issue: 

https://github.com/rancher/rancher/issues/47371
 
### Automated Testing

* Test types added/modified:
    * Unit
    * Integration (Go Framework)
    * Integration (v2prov Framework)
    * Validation (Go Framework)
    * Other - Explain: _EXPLAIN_
    * None
    * _REMOVE NOT APPLICABLE BULLET POINTS ABOVE_
* If "None" - Reason: _EXPLAIN THE REASON_
  
* If "None" - GH Issue/PR: _LINK TO GH ISSUE/PR TO ADD TESTS_

Summary: _TODO_

### Regressions Considerations

_TODO_

Existing / newly added automated tests that provide evidence there are no regressions:
* _TODO_

### CODE REVIEW

File: pkg/auth/api/secrets/cleanup_test.go [hunk 1]
```diff
@@ -135,58 +141,60 @@ func TestCleanupDeprecatedSecretsKnownConfig(t *testing.T) {
...
 
-	s, err = secrets.GetNamespaced(common.SecretsNamespace, secretName, metav1.GetOptions{})
+	s, err = secrets.Get(common.SecretsNamespace, secretName, metav1.GetOptions{})
 	assert.Errorf(t, err, "expected to not find the secret %s belonging to the disabled auth provider", secretName)
 	assert.Nil(t, s, "expected the secret to be nil")
 }
 
-func getSecretInterfaceMock(store map[string]*corev1.Secret) v1.SecretInterface {
-	secretInterfaceMock := &fakes.SecretInterfaceMock{}
+func getSecretInterfaceMock(ctrl *gomock.Controller, store map[string]*corev1.Secret) wcorev1.SecretController {
```
Comment (alegrey91) [inline pkg/auth/api/secrets/cleanup_test.go:155]: move from `getSecretInterfaceMock` to `getSecretControllerMock`?

Comment (crobby) [reply pkg/auth/api/secrets/cleanup_test.go:155]: sounds reasonable, I'll make that tweak

**Review by alegrey91 — APPROVED** [review]
Great job! LGTM

File: pkg/agent/clean/adunmigration/ldap.go [hunk 2]
```diff
@@ -2,28 +2,30 @@ package adunmigration
...
+	"context"
 	"crypto/x509"
 	"fmt"
 	"os"
 	"regexp"
 	"strings"
 	"time"
 
+	"github.com/rancher/rancher/pkg/wrangler"
+
```
Comment (JonCrowther) [inline pkg/agent/clean/adunmigration/ldap.go:14]: Can you get rid of the white space here?

File: pkg/auth/api/secrets/cleanup_test.go (hunk 1 above)
Comment (JonCrowther) [reply pkg/auth/api/secrets/cleanup_test.go:155]: Looks like it didn't get changed to `getSecretControllerMock`?

File: pkg/auth/cleanup/service_test.go [hunk 3]
```diff
@@ -9,13 +9,15 @@ import (
 	"testing"
 	"time"
 
+	wcorev1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+
```
Comment (JonCrowther) [inline pkg/auth/cleanup/service_test.go:13]: Could you get rid of this whitespace?

File: pkg/auth/providers/common/password.go
Comment (JonCrowther) [inline pkg/auth/providers/common/password.go:1]: On line 20, we still use the `corev1` version of `Secret`. Could you change it to `v1.Secret`? Looking at [sourcegraph](https://sourcegraph.com/search?q=context%3Aglobal+repo%3Agithub.com%2Francher+content%3A%22NameForSecret%22&patternType=standard&sm=0), it's only referenced in rancher/rancher so it should be ok to change the signature.

File: pkg/auth/providers/common/usermanager_test.go
Comment (JonCrowther) [inline pkg/auth/providers/common/usermanager_test.go:1]: Thank you for adding tests!

File: pkg/auth/providers/ldap/ldap_client_test.go [hunk 4]
```diff
@@ -8,13 +8,14 @@ import (
 	"testing"
 	"time"
 
+	wcorev1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+
```
Comment (JonCrowther) [inline pkg/auth/providers/ldap/ldap_client_test.go:12]: Could you get rid of the whitespace? Here and line 15

File: pkg/auth/providers/ldap/ldap_provider_test.go [hunk 5]
```diff
@@ -6,10 +6,11 @@ import (
 	"reflect"
 	"testing"
 
+	wcorev1 "github.com/rancher/wrangler/v3/pkg/generated/controllers/core/v1"
+
```
Comment (JonCrowther) [inline pkg/auth/providers/ldap/ldap_provider_test.go:10]: Please remove whitespace

File: pkg/auth/server.go [hunk 6]
```diff
@@ -56,6 +56,8 @@ func NewServer(ctx context.Context, cfg *rest.Config, wContext *wrangler.Context
 		return nil, err
 	}
 
+	sc.Wrangler = wContext
```
Comment (JonCrowther) [inline pkg/auth/server.go:59]: I have nothing against it, but is there a reason you moved setting the Wrangler context earlier here?

**Review by JonCrowther — COMMENTED** [review]
Looking good, just a couple nits and questions

**Review by JonCrowther — APPROVED** [review]
LGTM

**Review by alegrey91 — APPROVED** [review]

**Pull request merged**